homeassistant/components/efergy/sensor.py

```diff
@@ -31,15 +46,45 @@
 DEFAULT_PERIOD = "year"
 DEFAULT_UTC_OFFSET = "0"
 
-SENSOR_TYPES = {
-    CONF_INSTANT: ["Energy Usage", POWER_WATT],
-    CONF_AMOUNT: ["Energy Consumed", ENERGY_KILO_WATT_HOUR],
-    CONF_BUDGET: ["Energy Budget", None],
-    CONF_COST: ["Energy Cost", None],
-    CONF_CURRENT_VALUES: ["Per-Device Usage", POWER_WATT],
-}
+SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
```

Comment:
Keep this as a dict to not have to do a double loop for each monitored condition.

Reply:
or loop through SENSOR_TYPES and check if desc.key is in monitored_conditions

Reply:
The trouble with the second suggestion I'm having is with the format of the current yaml
```
monitored_variables:
  - type: current_values
  - type: instant_readings
  - type: amount
    period: day
  - type: amount
    period: month

Reply:
Then do the first.

Reply:
Renamed PR to not describe as a rework.

Reply:
You should still use `SensorEntityDescription`